### ISSUE TITLE
Unable to run yolov8 on Windows 11 GPU

### ISSUE THREAD

**Issue by haiph-dev** [ultralytics/ultralytics#5156]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Dear Ultralytics,

I'm sorry if I made a wrong question. I'm unable to run yolov8 on Windows 11 GPU. Please help to advise if I missed anything on your docs.

Thank you for your support.

### ultralytics.checks()
Ultralytics YOLOv8.0.107  Python-3.10.13 torch-2.0.1+cpu CPU
Setup complete  (32 CPUs, 127.9 GB RAM, 331.9/465.0 GB disk)

### environment setup:
CPU: AMD Ryzen 9 5950X
GPU: NVIDIA RTX 3070 Ti

### nvidia-smi
Fri Sep 29 14:49:22 2023
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 522.25       Driver Version: 522.25       CUDA Version: 11.8     |
|-------------------------------+----------------------+----------------------+
| GPU  Name            TCC/WDDM | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|                               |                      |               MIG M. |
|===============================+======================+======================|
|   0  NVIDIA GeForce ... WDDM  | 00000000:08:00.0  On |                  N/A |
|  0%   34C    P8    19W / 310W |    904MiB /  8192MiB |     10%      Default |
|                               |                      |                  N/A |
+-------------------------------+----------------------+----------------------+

### nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2022 NVIDIA Corporation
Built on Wed_Sep_21_10:41:10_Pacific_Daylight_Time_2022
Cuda compilation tools, release 11.8, V11.8.89
Build cuda_11.8.r11.8/compiler.31833905_0

### conda list

# Name                    Version                   Build  Channel
blas                      1.0                         mkl
brotlipy                  0.7.0           py310h2bbff1b_1002
bzip2                     1.0.8                he774522_0
ca-certificates           2023.08.22           haa95532_0
certifi                   2023.7.22       py310haa95532_0
cffi                      1.15.1          py310h2bbff1b_3
chardet                   5.2.0                    pypi_0    pypi
charset-normalizer        3.2.0                    pypi_0    pypi
colorama                  0.4.6                    pypi_0    pypi
contourpy                 1.1.1                    pypi_0    pypi
cryptography              41.0.3          py310h89fc84f_0
cuda-cccl                 12.2.140                      0    nvidia
cuda-cudart               11.8.89                       0    nvidia
cuda-cudart-dev           11.8.89                       0    nvidia
cuda-cupti                11.8.87                       0    nvidia
cuda-libraries            11.8.0                        0    nvidia
cuda-libraries-dev        11.8.0                        0    nvidia
cuda-nvrtc                11.8.89                       0    nvidia
cuda-nvrtc-dev            11.8.89                       0    nvidia
cuda-nvtx                 11.8.86                       0    nvidia
cuda-profiler-api         12.2.140                      0    nvidia
cuda-runtime              11.8.0                        0    nvidia
cudatoolkit               11.8.0               hd77b12b_0
cycler                    0.11.0                   pypi_0    pypi
filelock                  3.12.4                   pypi_0    pypi
fonttools                 4.42.1                   pypi_0    pypi
freetype                  2.12.1               ha860e81_0
giflib                    5.2.1                h8cc25b3_3
idna                      3.4             py310haa95532_0
intel-openmp              2023.1.0         h59b6b97_46319
jinja2                    3.1.2           py310haa95532_0
jpeg                      9e                   h2bbff1b_1
kiwisolver                1.4.5                    pypi_0    pypi
lerc                      3.0                  hd77b12b_0
libcublas                 11.11.3.6                     0    nvidia
libcublas-dev             11.11.3.6                     0    nvidia
libcufft                  10.9.0.58                     0    nvidia
libcufft-dev              10.9.0.58                     0    nvidia
libcurand                 10.3.3.141                    0    nvidia
libcurand-dev             10.3.3.141                    0    nvidia
libcusolver               11.4.1.48                     0    nvidia
libcusolver-dev           11.4.1.48                     0    nvidia
libcusparse               11.7.5.86                     0    nvidia
libcusparse-dev           11.7.5.86                     0    nvidia
libdeflate                1.17                 h2bbff1b_0
libffi                    3.4.4                hd77b12b_0
libnpp                    11.8.0.86                     0    nvidia
libnpp-dev                11.8.0.86                     0    nvidia
libnvjpeg                 11.9.0.86                     0    nvidia
libnvjpeg-dev             11.9.0.86                     0    nvidia
libpng                    1.6.39               h8cc25b3_0
libtiff                   4.5.1                hd77b12b_0
libuv                     1.44.2               h2bbff1b_0
libwebp                   1.3.2                hbc33d0d_0
libwebp-base              1.3.2                h2bbff1b_0
lz4-c                     1.9.4                h2bbff1b_0
markupsafe                2.1.3                    pypi_0    pypi
matplotlib                3.8.0                    pypi_0    pypi
mkl                       2023.1.0         h6b88ed4_46357
mkl-service               2.4.0           py310h2bbff1b_1
mkl_fft                   1.3.8           py310h2bbff1b_0
mkl_random                1.2.4           py310h59b6b97_0
mpmath                    1.3.0           py310haa95532_0
networkx                  3.1             py310haa95532_0
numpy                     1.26.0          py310h055cbcc_0
numpy-base                1.26.0          py310h65a83cf_0
openssl                   3.0.11               h2bbff1b_2
packaging                 23.1                     pypi_0    pypi
pandas                    2.1.1                    pypi_0    pypi
pillow                    10.0.1                   pypi_0    pypi
pip                       23.2.1          py310haa95532_0
psutil                    5.9.5                    pypi_0    pypi
pycparser                 2.21               pyhd3eb1b0_0
pyopenssl                 23.2.0          py310haa95532_0
pyparsing                 3.1.1                    pypi_0    pypi
pysocks                   1.7.1           py310haa95532_0
python                    3.10.13              he1021f5_0
python-dateutil           2.8.2                    pypi_0    pypi
pytorch                   2.0.1           py3.10_cuda11.8_cudnn8_0    pytorch
pytorch-cuda              11.8                 h24eeafa_5    pytorch
pytorch-mutex             1.0                        cuda    pytorch
pytz                      2023.3.post1             pypi_0    pypi
pyyaml                    6.0.1                    pypi_0    pypi
requests                  2.31.0          py310haa95532_0
scipy                     1.11.3                   pypi_0    pypi
seaborn                   0.12.2                   pypi_0    pypi
setuptools                68.0.0          py310haa95532_0
six                       1.16.0                   pypi_0    pypi
sqlite                    3.41.2               h2bbff1b_0
sympy                     1.12                     pypi_0    pypi
tbb                       2021.8.0             h59b6b97_0
tk                        8.6.12               h2bbff1b_0
torchaudio                2.0.2                    pypi_0    pypi
torchvision               0.15.2                   pypi_0    pypi
tqdm                      4.66.1                   pypi_0    pypi
typing-extensions         4.8.0                    pypi_0    pypi
typing_extensions         4.7.1           py310haa95532_0
tzdata                    2023.3                   pypi_0    pypi
urllib3                   1.26.16         py310haa95532_0
vc                        14.2                 h21ff451_1
vs2015_runtime            14.27.29016          h5e58377_2
wheel                     0.41.2          py310haa95532_0
win_inet_pton             1.1.0           py310haa95532_0
xz                        5.4.2                h8cc25b3_0
zlib                      1.2.13               h8cc25b3_0
zstd                      1.5.5                hd43e919_0

**Comment by RizwanMunawar** [comment]
Hello @haiph-dev, Welcome to the Ultralytics Community 🤗

To run YOLOv8 on Windows 11, you can follow the steps below:

1. Install the Ultralytics package by running the following command in your terminal or command prompt:

   ```
   pip install ultralytics
   ```

2. After the installation is complete, you can use the following command to start detection using YOLOv8:

   ```
   yolo task=detect mode=predict source="https://www.youtube.com/watch?v=iSXPMH3Wfig&t=19s" show=True save=True
   ```

This command will initiate the YOLOv8 model to perform object detection on the YouTube video specified in the `source` parameter. It will display the results with `show=True` and save the output with `save=True`.

Make sure you have the required dependencies and libraries installed before running these commands. If you encounter any issues or have further questions, please feel free to ask.

Hope this helps! Let me know if you have any additional questions.
Thanks

**Comment by haiph-dev** [comment]
Dear Rizwan,

I tried all the ways I found including installing on different conda environment but yolov8 only running cpu:
conda install -c pytorch -c nvidia -c conda-forge pytorch torchvision pytorch-cuda=11.8 ultralytics 

Please help to refer my details packages above and advise how can I troubleshoot what missing.
Thank you for your prompt answer

**Comment by glenn-jocher** [comment]
Dear @haiph-dev,

I'm sorry to hear that you're having trouble running YOLOv8 on GPU. Based on the information you provided, it seems like you have already installed the necessary packages, including PyTorch, torchvision, and the GPU compatible version of PyTorch.

Here are a few things you can check to help troubleshoot the issue:

1. **Verify CUDA Compatibility and Installation:** Make sure your GPU is CUDA compatible and you have correctly installed CUDA on your machine. You can check this by running the 'nvidia-smi' command in your terminal, which should show the status of your NVIDIA GPU and CUDA version.

2. **Check PyTorch and CUDA Integration:** You want to ensure that PyTorch can successfully utilize your CUDA. A simple way to do this is to open a Python terminal and run 'import torch; print(torch.cuda.is_available())'. If this returns 'True', then PyTorch is successfully set up to use CUDA.

3. **Ensure Correct Environment Activation:** Make sure you are working in the right environment where you installed the packages. Sometimes, packages installed in a specific Conda environment may not be accessible when you are working in a different one. Always double-check whether your desired environment is active.

4. **Update Your Packages:** It's possible that outdated versions of certain packages may not be compatible with your GPU. Ensure that all of your packages are up-to-date.

Lastly, and it's less likely but check if the program or code you're trying to execute is set to run on a GPU device. In YOLOv8, this would probably be a settings or configuration option.

I hope this helps! Let us know how it goes or if you have more questions.

**Comment by haiph-dev** [comment]
Dear @glenn-jocher,

I tried to run docker image 'ultralytics/ultralytics:latest' and mount project folder on host to container.
Same code and it run with my GPU from container, so I just go with this way until I try find another way back with Windows GPU later. It would be better if you have tips to troubleshoot this issue on Windows.

root@f3652578ddbb:/aisec-yolov8#  cd /aisec-yolov8 ; /usr/bin/env /opt/conda/bin/python /root/.vscode-server/extensions/ms-python.python-2023.16.0/pythonFiles/lib/python/debugpy/adapter/../../debugpy/launcher 49209 -- /aisec-yolov8/test.py 
Ultralytics YOLOv8.0.188 ?? Python-3.10.11 torch-2.0.1 CUDA:0 (NVIDIA GeForce RTX 3070 Ti, 8192MiB)
Setup complete ? (32 CPUs, 62.7 GB RAM, 35.0/250.9 GB disk)
